slsim/lens_pop.py

```diff
@@ -162,18 +180,37 @@ def __init__(
             # develop a supernovae class inside the slsim and them here to generate
             # supernovae light curves.
             self.path = os.path.dirname(__file__)
-            new_path = self.path + "/Sources/SupernovaeData/supernovae_data.pkl"
-            with open(new_path, "rb") as f:
-                load_supernovae_data = pickle.load(f)
-            self._sources = PointPlusExtendedSources(
-                load_supernovae_data,
-                cosmo=cosmo,
-                sky_area=sky_area,
-                kwargs_cut=kwargs_source_cut,
-                variability_model=variability_model,
-                kwargs_variability_model=kwargs_variability,
-                list_type="list",
-            )
+            if catalog_type == "scotch":
+                new_path = self.path + "/Sources/SupernovaeData/scotch_host_data.fits"
```

Comment:
do you want to provide an option to explicitly set a path to the catalogue, especially since the one you have is a small subset of it?